api/cloud/oc_cloud_context.c

```diff
@@ -110,6 +111,7 @@ cloud_context_clear(oc_cloud_context_t *ctx)
   memset(ctx->cloud_ep, 0, sizeof(oc_endpoint_t));
   ctx->cloud_ep_state = OC_SESSION_DISCONNECTED;
   cloud_manager_stop(ctx);
+  cloud_deregister_stop(ctx);
```

Comment:
And what about when  someone set new cloud during registration? Do we have covered this usecase?
I think we need add somewhere this `cloud_deregister_stop`.

Reply:
Should be handled now, I've added `cloud_deregister_stop` after each call of `cloud_close_endpoint` (which timeouts all request handlers) in `api/cloud` files.